discord_connector.py

```diff
@@ -158,128 +162,183 @@ def _handle_test_connectivity(self, param):
-            # for now the return is commented out, but after implementation, return from here
             return action_result.get_status()
-            pass
 
-        # Now post process the data,  uncomment code as you deem fit
-
-        # Add the response into the data section
         action_result.add_data(response)
-
-        # Add a dictionary that is made up of the most important values from data into the summary
         summary = action_result.update_summary({})
         summary['num_channels'] = len(response)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_fetch_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        attachments, embeds = None, None
+        message = self._async_loop.run_until_complete(self.fetch_message(channel_id, message_id))
+
+        # do we need to check if message is not none?
+        if message is not None:
+            if message.embeds is not None or message.attachments is not None:
+                attachments, embeds = self.create_artifacts(message)
+            message = self.parse_message(message, attachments, embeds)
+        else:
+            summary = action_result.update_summary({})
+            summary['failure: '] = "unable to fetch the message: message is None"
+            return action_result.set_status(phantom.APP_ERROR)
+
+        action_result.add_data(message)
+        summary = action_result.update_summary({})
+        summary['success: '] = "fetching message completed"
 
-        # Return success, no need to set the message, only the status
-        # BaseConnector will create a textual message based off of the summary dictionary
         return action_result.set_status(phantom.APP_SUCCESS)
 
-        # For now return Error with a message, in case of success we don't set the message, but use the summary
-        #return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
+    async def fetch_message(self, channel_id, message_id):
+        await self._client.login(self._token)
+
+        guild = await self._client.fetch_guild(self._guild_id)
+        self.save_progress("fetched guild: {}, fetched guild id: {}".format(str(guild), str(guild.id)))
+
+        channel = await guild.fetch_channel(channel_id)
+        self.save_progress("channel: {}".format(str(channel)))
+
+        message = await channel.fetch_message(message_id)
+        self.save_progress("message: {}".format(str(message)))
+
+        await self._client.close()
+        return message
+
+    def create_artifacts(self, message):
+
+        container_id = BaseConnector.get_container_id(self)
+        attachments = []
+        embeds = []
+
+        # do we need it there or rather in consts???
+        artifact = {
```

Comment:
What artifact?

Reply:
It used to be a blueprint containing artifact structure, it is unused now